src/main/java/WayofTime/bloodmagic/compress/AdvancedCompressionHandler.java

```diff
@@ -4,12 +4,82 @@
 import net.minecraft.inventory.Container;
 import net.minecraft.inventory.InventoryCrafting;
 import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.CraftingManager;
+import net.minecraft.item.crafting.IRecipe;
 import net.minecraft.world.World;
 
+
 public class AdvancedCompressionHandler extends CompressionHandler {
+    private static ItemStack reversibleCheck;
+
+    public static boolean isResultStackReversible(ItemStack stack, int gridSize, World world) {
+
+        if (stack.isEmpty()) {
+            return false;
+        }
+        InventoryCrafting inventory = new InventoryCrafting(new Container() {
```

Comment:
This should be stored as a constant. No need to create a new instance each time this is called.